### PR TITLE
BITAU-200 Log non-fatal authenticator bridge errors

### PR DESCRIPTION
**Pull request by ahaisting-livefront**
## 🎟️ Tracking

https://livefront.atlassian.net/browse/BITAU-200

## 📔 Objective

Log a few non-fatal errors in the authenticator bridge sync process.

## 📸 Screenshots

N/A

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/data/platform/util/AuthenticatorBridgeException.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.x8bit.bitwarden.data.platform.util
+
+/**
+ * Represents a non-fatal authenticator bridge exception.
+ */
+class AuthenticatorBridgeException : Exception()
```
Comment (ahaisting-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/platform/util/AuthenticatorBridgeException.kt:6]: So I'm not sure this is even needed. I could just new up some kind an `IllegalStateException` or even just an `Exception` instead. I've done this in the past and I remember it helped on the filtering side in the crashlytics console.

Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/data/platform/util/AuthenticatorBridgeException.kt:6]: 🪙 🪙 Passing an `IllegalStateException` with a descriptive message achieves the same thing with less overhead.

File: app/src/main/java/com/x8bit/bitwarden/data/platform/processor/AuthenticatorBridgeProcessorImpl.kt [hunk 2]
```diff
@@ -110,14 +117,33 @@ class AuthenticatorBridgeProcessorImpl(
...
+                    message = "Unable to start add TOTP item flow when " +
+                        "symmetricEncryptionKeyData is null.",
+                )
+                return false
+            }
             val intent = createAddTotpItemFromAuthenticatorIntent(context = applicationContext)
             val totpData = data.decrypt(symmetricEncryptionKey)
+                .onFailure {
+                    Timber.e(
+                        t = IllegalStateException(),
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/platform/processor/AuthenticatorBridgeProcessorImpl.kt:132]: Can we use the `it` error here?

File: app/src/main/java/com/x8bit/bitwarden/data/platform/processor/AuthenticatorBridgeProcessorImpl.kt [hunk 3]
```diff
@@ -110,14 +117,33 @@ class AuthenticatorBridgeProcessorImpl(
...
+                }
                 .getOrNull()
                 ?.totpUri
                 ?.toUri()
                 ?.getTotpDataOrNull()
-                ?: return false
+                ?: run {
+                    Timber.e(
+                        t = IllegalStateException(),
+                        message = "Unable to parse TOTP URI.",
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/platform/processor/AuthenticatorBridgeProcessorImpl.kt:143]: So this error _could_ get logged twice, once here and possibly once if the decryption was the error. Is that the intention?

Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/data/platform/processor/AuthenticatorBridgeProcessorImpl.kt:143]: Unlikely. Adding a `return` to the `onFailure` block.

**Review by david-livefront — APPROVED** [review]

**Pull request merged**